File: backend/app/schemas/direct_message.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class DirectMessageCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=4000)
    reply_to_id: Optional[int] = None


class DirectMessageUpdate(BaseModel):
    content: str = Field(..., min_length=1, max_length=4000)


class DirectMessageRead(BaseModel):
    is_read: bool = True


class DirectMessageResponse(BaseModel):
    id: int

    conversation_id: int

    sender_id: int

    content: str

    reply_to_id: Optional[int]

    is_read: bool
    is_edited: bool
    is_deleted: bool

    created_at: datetime
    edited_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)


class DirectMessageListResponse(BaseModel):
    messages: list[DirectMessageResponse]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/direct-message.ts
export interface DirectMessage {
    id: number;

    conversation_id: number;

    sender_id: number;

    content: string;

    is_read: boolean;

    is_deleted: boolean;

    is_edited: boolean;

    created_at: string;

    edited_at?: string;
}

export interface DirectMessageCreate {
    content: string;
}

export interface DirectMessageUpdate {
    content: string;
}

export interface DirectMessageListResponse {
    messages: DirectMessage[];
}